nuxeo-features/nuxeo-annotation/nuxeo-annotation-core/src/main/java/org/nuxeo/ecm/annotation/AnnotationServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ *
+ */
+
+package org.nuxeo.ecm.annotation;
+
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_COLOR_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_CREATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOCUMENT_ID_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOC_TYPE;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_FLAGS_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_ID_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_LAST_MODIFIER_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_NAME_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_OPACITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_PAGE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_POSITION_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SECURITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SUBJECT_PROPERTY;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @since 10.1
+ */
+public class AnnotationServiceImpl extends DefaultComponent implements AnnotationService {
+
+    private static final Log log = LogFactory.getLog(AnnotationServiceImpl.class);
+
+    protected static final String ANNOTATIONS_FROM_DOCUMENT = "SELECT * FROM Annotation WHERE annotation:annotationId LIKE '%s%%' ";
+
+    @Override
+    public String createAnnotation(DocumentModel documentModel, Annotation annotation) {
+        CoreSession session = documentModel.getCoreSession();
+        String id = documentModel.getRepositoryName() + "-" + documentModel.getId() + "-" + annotation.getDocumentId();
+
+        // Create annotation as a placeless document
+        DocumentModel annotationModel = session.createDocumentModel(null, id, ANNOTATION_DOC_TYPE);
```

Comment:
You should call this `name`, not `id`. And we can probably give it a fixed value (like `"annotation"`) as there will be no query on it that I can see.